File: 217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        mp = {}
        n = len(nums)
        i=0
        while i<n and mp.get(nums[i],0)<1:
            mp[nums[i]] = mp.get(nums[i],0)+1
            i+=1
        if i==n:
            return False
        else:
            return True

File: 217. Contains Duplicate.java
import java.util.HashMap; //Via HashMap
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        int i=0;
        for(int num: nums){
            if(map.containsKey(num) && map.get(num) >= 1) return true;
            map.put(num,map.getOrDefault(num,0)+1);
        }
        return false;
       
    }
}


import java.util.HashSet;  //Via HashSet
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for (int num : nums) {
            if (set.contains(num))
                return true;
            set.add(num);
        }
        return false;
    }
}
